nixos/modules/services/web-apps/powerdns-admin.nix

```
@@ -0,0 +1,147 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.powerdns-admin;
+
+  configText = ''
+    ${cfg.config}
+  ''
+  + optionalString (cfg.secretKeyFile != null) ''
+    with open('${cfg.secretKeyFile}') as file:
```

```suggestion
    with open('${cfg.secretKeyFile}', 'r') as file:
```

The service cannot and should not be able to write to the file (same for `saltFile`).